From fbe279155723a409bebef19a1beff26c5dc382fb Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Sat, 23 Mar 2024 01:46:09 -0600
Subject: [PATCH] std/math/hardware.d: use an alternative register naming ...

... for LoongArch64 so that LDC/LLVM can recognize certain fp
instructions
---
 std/math/hardware.d | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/std/math/hardware.d b/std/math/hardware.d
index f474ea903..4bfe79242 100644
--- a/runtime/phobos/std/math/hardware.d
+++ b/runtime/phobos/std/math/hardware.d
@@ -203,7 +203,7 @@ private:
         {
             asm nothrow @nogc
             {
-                "movgr2fcsr $r2,$r0";
+                "movgr2fcsr $fcsr2,$r0";
             }
         }
         else
@@ -933,7 +933,7 @@ private:
         {
             asm nothrow @nogc
             {
-                "movgr2fcsr $r0,%0" :
+                "movgr2fcsr $fcsr0,%0" :
                 : "r" (newState & (roundingMask | allExceptions));
             }
         }
-- 
2.44.0

